helpers/providers/provider.py

```diff
@@ -0,0 +1,18 @@
+''' Contains the abstract class(interface) for providers '''
+from abc import abstractmethod, ABC
+
+
+class Provider(ABC):
```

Comment:
I don't quite understand the purpose of this class. It looks like your intent is to use it as an abstract class (hence ABC) that defines all the methods that a Provider needs to implement. But at the same time you declare all methods to be class methods (`@classmethod`) which looks like an antipattern.

Contracts like this will lead to global state which may be hard to debug.

As an example, in `provider_test.py` you do

```
self.provider = GitHubHTTPProvider
```

which just references the class `GitHubHTTPProvider` itself instead of an instance of it (that would be `GitHubHTTPProvider()`).

Reply:
Yep. Exactly